### ISSUE TITLE
User responses to mandatory tool reservation questions are not always recorded

### ISSUE THREAD

**Issue by enodev0** [usnistgov/NEMO#170]
Hi there,

Thanks for this awesome tool! So our organization has been using NEMO since around 2021 and recently we noticed a weird issue which is described below. Any hints or suggestions is greatly appreciated!

### Observation
After a user reserves a working time slot for a tool from the calendar view, the question responses are not always visible when clicking on the reserved slot. 

Only a handful of basic info like **user, created by, created on, start, end ..** and so on are visible, even though the reservation form asks many more mandatory questions.

### Expected observation
When clicking on a reserved slot, the user responses to the reservation questions should be visible.

### How to replicate
Seems to occur randomly, does not appear to be linked to any particular tool, user or time slot.

### Attempts to diagnose the issue
Our NEMO installation uses SQLite, so we started to poke around the contents of the **.db** file. Looking at the **NEMO_reservations** table, we have had around ~ 10k reservations since we set up the system in 2021. The **question_data** column seems to be storing the user responses. 

We found that some entries in the table have the **question_data** column set to NULL, even though the questions are mandatory.

Running the command
```{sql}
select * from NEMO_reservation where question_data is NULL and cancelled == 0
```
returns ~2k entries where the responses were not saved. Looking at the **user_id, project_id and tool_id** columns yields a mixed bag with no clear patterns.

### Software Versions
NEMO:  v3.12.2, running from a docker container on a Debian 10 VM.

**Comment by rptmat57** [comment]
Hi,
You are running an older version of NEMO, and I suspect your issue is link to #104 which was fixed in v3.15.0

An easy way to check would be to create a reservation, click on it to confirm the questions are there, then move the reservation in the calendar, and try opening it again, and confirm that this time the question data was not kept.

If that's the case (and even if it's not) I strongly recommend updating to the latest version of NEMO to see if the issue still happens.

Also, I am glad to hear you have been using NEMO since 2021!
I would really appreciate if you wouldn't mind sending me a quick email at mathieu.rampant@nist.gov to let me know what your organization does and how you are using NEMO, as we do not have any visibility at all on who is actually using it and it really helps us justify the resources spent on maintaining this project.

Thanks!

**Comment by enodev0** [comment]
Well your diagnosis was spot on! The issue can be reproduced by moving the reservations.

We'll be moving over to the current version of NEMO.

Your efforts are much appreciated! Thanks a lot.
